waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java

```diff
@@ -88,7 +88,15 @@ public MetaStoreFilterHook getMetastoreFilter() {
 
   @Override
   public Table transformOutboundTable(Table table) {
-    table.setDbName(metaStoreMapping.transformOutboundDatabaseName(table.getDbName()));
+    String originalDatabaseName = table.getDbName();
+    String databaseName = metaStoreMapping.transformOutboundDatabaseName(originalDatabaseName);
+    table.setDbName(databaseName);
+    if (databaseName.equalsIgnoreCase(originalDatabaseName)) {
```

Comment:
I don't understand what's going on here - why is the db name the same in this case?

Reply:
I don't know if it is that's why I'm checking:
if our transform didn't change the databasename we don't need to try and do the parsing just return the table as is (nothing would have changed).

This creeped in when we removed the IdentityMapping (which was just returning table for non prefixed metastores). So the if will mimic that behaviour again.